features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Documents/Listing.jsx

```diff
@@ -105,29 +108,48 @@ class Listing extends React.Component {
      Get the document list attached to current API and set it to the state
      */
     getDocumentsList() {
-        const api = new API();
+        const { apiType } = this.props.api;
         const { intl } = this.props;
-        const docs = api.getDocuments(this.props.api.id);
-        docs.then((response) => {
-            this.setState({ docs: response.obj.list });
-        }).catch((errorResponse) => {
-            const errorData = JSON.parse(errorResponse.message);
-            const messageTxt =
-                'Error[' + errorData.code + ']: ' + errorData.description + ' | ' + errorData.message + '.';
-            console.error(messageTxt);
-            Alert.error(intl.formatMessage({
-                id: 'Apis.Details.Documents.Listing.documents.listing.fetching.error.message',
-                defaultMessage: 'Error in fetching documents list of the API',
-            }));
-        });
+
+        if (apiType === API.CONSTS.APIProduct) {
+            const apiProduct = new APIProduct();
+            const docs = apiProduct.getDocuments(this.props.api.id);
```

Comment:
use destructuring for `this.props.api`

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/6759/commits/0c218762d38e7db55b9b2b6306a4a1027e6d4cb0